js/views/userPageVw.js

```diff
@@ -1444,6 +1467,70 @@ module.exports = baseVw.extend({
 
   },
 
+  addModeratorClick: function(e){
+    var $targ = $(e.target).closest('.js-addmoderator'),
+        self = this,
+        modList = {};
+
+    $targ.addClass('loading');
+
+    modList.moderators = this.model.get('user').moderator_guids;
+    modList.moderators.push(this.pageID);
+
+    saveToAPI('', this.model.get('user'), this.model.get('user').serverUrl + "settings",
+      function(){
+        // confirmed
+        self.options.userModel.fetch({
+          success: function(model, response) {
+            if (self.isRemoved()) return;
+            var user = self.model.get('user');
+            user.moderators = model.get('moderators');
+            user.moderator_guids = model.get('moderator_guids');
+            self.getIsModerator();
+          }
+        });
+      }, '', modList,'', '').always(function(){
+        $targ.removeClass('loading');
+      });
+  },
+
+  removeModeratorClick: function(e){
+    var $targ = $(e.target).closest('.js-removemoderator'),
+        self = this,
+        modList = {};
+
+    if($targ.hasClass('confirm')){
+      $targ.addClass('loading').removeClass('confirm');
+
+      modList.moderators = [];
+
+      __.each(this.model.get('user').moderator_guids, function(mod) {
+        if(mod != self.pageID && (typeof(mod) !== 'object' || mod.guid != self.pageID)) {
+          modList.moderators.push(mod);
+        }
+      });
```

Comment:
It would be better to do this by getting the index of this.pageID in moderator_guids and using splice to remove it. Or use underscore's without to be very concise. http://underscorejs.org/#without

That way you wouldn't need to do any looping.